### DISCUSSION TITLE
Combine cookie authentication with openidc on the same location ?

### DISCUSSION THREAD

**Discussion by shrirangangmail** [OpenIDC/mod_auth_openidc#1024] [Q&A]
Hello, first time working with apache httpd and mod_auth_openidc so this may be a bad or trivial  question, but could not find anything after searching - we have a site where we have to provide two ways of auth - one with openidc and for a select few users who are outside the provider scope, a cookie based authentication. Both should be able to access the same resources once authenticated.
Is it possible at all ?  I tried `<If ><Else>` within the directory directive but does not seem to work.  
we are using apache 2.4.54.
`<Directory "/home/ws/proj/cgi-bin">`
`  <If "%{HTTP_COOKIE} =~ /SWMAUTH/">`
   `   SetEnv ALLOWACCESS 1`
  `   Require env ALLOWACCESS`
 ` </If>`
`  <Else>`
   `  AuthType openid-connect`
  `   Require valid-user`
 `  </Else>`
`</Directory>`

 TIA !

**Comment by zandbelt** [comment]
I don't think if/then/else would work because authentication handlers (like mod_auth_openidc) are executed before any of that logic is applied; you should be able to use 2 virtual hosts , each with their own AuthType, and proxy between those

**Reply by shrirangangmail** [reply]
Thank you for the response. I will try it out if I can (need to convince the admin) . For now, I used requireAny directive and that seems to be sufficient for our purposes.
SetEnvIf Cookie "SWMAUTH=" ALLOWACCESS=1
<Directory "/home/ws/proj/cgi-bin">
    <RequireAny>
         Require env ALLOWACCESS
         AuthType openid-connect
         Require valid-user
   </RequireAny>
</Directory>
